[Shauryan123] iP

Shauryan123 opened the pull request:
### DukePro - ShaunBot
**“Your mind is for having ideas, not holding them.” – David Allen**

DukePro frees your mind of having to remember things you need to do. It's:

- visual-based
- easy to learn
- FAST SUPER FAST to use

All you need to do is:

1. download it from [here](https://github.com/Shauryan123/ip/releases/tag/A-JUnit).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😉
5. And it is FREE!

Features:

- [X]  Managing tasks
- [X]  Managing deadlines 
- [X]  Reminders

Commit:
Add Gradle support

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

Eclipse-Dominator committed:
Bump gradle and lib version

Commit:
docs/README.md: Tweak document template

Let's tweak the docs/README.md (which is used as the user guide)
to fit Duke better. Specifically,

1. mention product name in the title
2. mention adding a product screenshot and a product intro
3. tweak the flow to describe feature-by-feature

Commit:
docs/README.md: Tweak document template

Let's tweak the docs/README.md (which is used as the user guide)
to fit Duke better. Specifically,

1. mention product name in the title
2. mention adding a product screenshot and a product intro
3. tweak the flow to describe feature-by-feature

Shauryan123 committed:
Level 0 commit

Shauryan123 committed:
Level-1 commit

Shauryan123 committed:
Level-2 commit

Shauryan123 committed:
Level-3 commit

Shauryan123 committed:
Level-4 commit

Shauryan123 committed:
Automated-UI-testing

Shauryan123 committed:
Level-5 commit

Shauryan123 committed:
Level-6 commit

Shauryan123 committed:
Level-7

Shauryan123 committed:
Level-7 refined code (added abstractions in Duke)

Shauryan123 committed:
Level-8 commit

Shauryan123 committed:
Applied OOP Changes

Shauryan123 committed:
A-Packages commit

Shauryan123 committed:
Merge branch 'add-gradle-support'

Shauryan123 committed:
A-Gradle Commit

Shauryan123 committed:
A-Gradle Commit

Shauryan123 committed:
Corrected A-packages changes

Shauryan123 committed:
corrected A-packages changes

Shauryan123 committed:
Added J-Unit Tests

Shauryan123 committed:
Added J-Unit Tests

Shauryan123 committed:
Added JavaDocs for half of the public methods/classes

Shauryan123 committed:
Added JavaDocs for half of the public methods/classes

Shauryan123 committed:
Made functionality for find and search

Shauryan123 committed:
Made functionality for find and search

Shauryan123 committed:
Refined Changes for Find functionality

Shauryan123 committed:
Refined Changes for Find functionality

Shauryan123 committed:
Made changes according to coding standard

Shauryan123 committed:
Made changes according to coding standard

Shauryan123 committed:
Add J-Unit Tests

Shauryan123 committed:
Add JavaDocs for half of the public methods/classes

Shauryan123 committed:
Make functionality for find and search

Shauryan123 committed:
Refine changes for find functionality

Shauryan123 committed:
Make changes according to coding standard

Shauryan123 committed:
Resolve Merge Conflicts from branch-Level-9

Shauryan123 committed:
Resolve merge conflicts from branch-A-CodingStandard

Shauryan123 committed:
Resolve merge conflicts from branch-Level-9

Shauryan123 committed:
Add GUI to chatbot for level 10

Shauryan123 committed:
Add images for chatbot GUI

Shauryan123 committed:
Add checkstyle configurations

taeewonnn commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,174 @@
+import javafx.scene.layout.VBox;
+import javafx.stage.Stage;
+import duke.task.TaskList;
+import duke.Ui.Ui;
```
Should the import statements be in groups? 🤔

taeewonnn commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,174 @@
+        scrollPane.setVvalue(1.0);
+        scrollPane.setFitToWidth(true);
+
+        // You will need to import `javafx.scene.layout.Region` for this.
```
I like how the comments are simple in explanation and clear to follow 😄

taeewonnn commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,105 @@
+     * @return The type of task the user wants to do.
+     */
+    public String identify(String command) {
+        String[] strings = command.split(" ");
```
I like that the naming follows the camelCase convention and has a plural form when necessary 👍

taeewonnn commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,105 @@
+        String result;
+        try {
+            switch (command) {
+                case "deadline":
```
Should the indentation be removed for case clauses?

taeewonnn commented on src/main/java/duke/Ui/MainWindow.java:
```diff
@@ -0,0 +1,77 @@
+        response = duke.getResponse(input);
+
+        dialogContainer.getChildren().addAll(
+                DialogBox.getUserDialog(input, userImage),
```
I like the layout of this method call 😃

taeewonnn reviewed (commented):
Overall, I found your code very easy to read and there are comments clear in explanations, allowing me to follow it easily. There are just a few minor coding standard violations that i noted and perhaps the start function in Duke.java can be very long. Regardless, it is a great work done!

Shauryan123 committed:
Add hover and pulse effects to make attractive GUI

Shauryan123 committed:
Refactor Chatbot GUI for Enhanced User Interaction with JavaFX Effects

The current GUI of the chatbot is functional yet lacks the visual appeal and interactive feedback that modern users expect. This limitation can hinder user engagement and overall user experience, making the interface feel outdated and less intuitive.

Update the GUI using JavaFX to incorporate pulse, hover, and slide effects. These effects are chosen to enrich the visual feedback and make the interaction more dynamic and engaging.

Pulse effects are introduced for immediate feedback on button clicks, hover effects are added to highlight interactive elements upon mouse-over, and slide effects are utilized for smooth transitions between chatbot sections. This approach is taken to not only enhance the aesthetic appeal but also to improve the intuitiveness and responsiveness of the user interface.

These changes are implemented with careful consideration of performance and accessibility. Testing is conducted extensively to ensure that the new effects are optimally integrated without compromising the application's speed or accessibility for users with different needs.

This update aims to significantly improve the user interaction model of the chatbot, making it more engaging and enjoyable to use, thereby potentially increasing user retention and satisfaction.

Shauryan123 committed:
Add streams

Shauryan123 committed:
Add B-ViewSchedules

Shauryan123 committed:
Merge branch 'extension'

Shauryan123 committed:
Add assertions

Shauryan123 committed:
Merge pull request #1 from Shauryan123/branch-A-Assertions

Add assertions for code relaibility

Shauryan123 committed:
refactor for code-quality

Shauryan123 committed:
Merge pull request #2 from Shauryan123/branch-A-CodeQuality

refactor for code-quality

Shauryan123 committed:
Add better GUI

Shauryan123 committed:
Merge branch 'master' of github.com:Shauryan123/ip

Shauryan123 committed:
add user guide

Shauryan123 committed:
Add Ui.png

Shauryan123 committed:
Update README.md

Shauryan123 committed:
Update README.md
